src/types/PlayerBlockAction.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+namespace pocketmine\network\mcpe\protocol\types;
+
+use pocketmine\network\mcpe\protocol\serializer\PacketSerializer;
+
+/** This is used for PlayerAuthInput packet when the flags include PERFORM_BLOCK_ACTIONS */
+final class PlayerBlockAction{
+
+	public int $actionType;
+	public BlockPosition $blockPos;
+	public int $face;
+
+	public function read(PacketSerializer $in) : void{
+		$this->actionType = $in->getVarInt();
+		if(match($this->actionType){
+			PlayerAction::ABORT_BREAK, PlayerAction::START_BREAK, PlayerAction::CRACK_BREAK, PlayerAction::PREDICT_DESTROY_BLOCK, PlayerAction::CONTINUE_DESTROY_BLOCK => true,
```

Comment:
This looks awful.

Reply:
The other 2 options are
```php
if($this->actionType !== PlayerAction::STOP_BREAK && $this->actionType >= PlayerAction::START_BREAK){}
```
or one with a bunch of `or`s

Reply:
I think a `switch` would be good enough here, to be honest.